src/components/dialog/dialog.svelte

```diff
@@ -42,6 +45,20 @@
     isOpen = false
     onClose?.()
   }
+
+  const setSize = (matches: boolean) => {
+    size = matches ? 'mobile' : 'normal'
+  }
+
+  onMount(() => {
```

Comment:
Maybe this should be a reactive statement so the media query updates when `mobileBreakpoint` is changed